src/internal_events/common.rs

```diff
@@ -115,11 +115,13 @@ impl InternalEvent for StreamClosedError {
             "error_type" => "stream_closed",
             "stage" => "sending",
         );
-        counter!(
-            "component_discarded_events_total", self.count as u64,
-            "error" => self.error.to_string(),
-            "error_type" => "stream_closed",
-            "stage" => "sending",
-        );
+        if self.count > 0 {
```

Comment:
I'd probably still publish the metric regardless for consistency, just with `0`.